src/ott/solvers/outputs.py

```diff
@@ -0,0 +1,66 @@
+from typing import Optional, Tuple
+
+import jax.numpy as jnp
+from flax import struct
+
+from ott.geometry import geometry
+from ott.problems.linear import linear_problem
+
+
+@struct.dataclass
+class _BaseTransportOutput:
+  """Base class for Transport output.
+
+  Args:
+    ot_prob: OT problem.
+    errors: Holds sequence of vectors of errors of the Sinkhorn algorithm
+      at each iteration.
+    shape: Shape of the transport plan.
+    costs: Holds the sequence of regularized GW costs seen through the loop
+      of the solver.
+    converged: Boolean value wether or not the problem has converged.
+    reg_ot_cost: Value of the regularised OT loss.
+  """
+
+  # TODO(michalk8): must be called `errors`, because of `store_inner_errors`
+  # in future, enforce via class hierarchy
+  ot_prob: linear_problem.LinearProblem
+  errors: jnp.ndarray
+  shape: Tuple[int, int] = struct.field(pytree_node=False)
+  # TODO(michalk8): Optional is an artifact of the current impl., refactor
+  converged: bool = False
+  costs: Optional[jnp.ndarray] = None
+  reg_ot_cost: Optional[float] = None
+
+  @property
```

Comment:
I don't think this is used anywhere, not sure why it was added.
Maybe when defining the abstraction for GW it will be useful, but for now, would remove it.